diag_manager/fms_diag_file_object.F90

```diff
@@ -183,6 +196,23 @@ logical function fms_diag_files_object_init ()
   fms_diag_files_object_init = .false.
 #endif
 end function fms_diag_files_object_init
+!> \brief Adds a field ID to the file
+subroutine add_field_id (obj, new_field_id)
+  class(fmsDiagFile_type), intent(inout) :: obj !< The file object
+  integer, intent(in) :: new_field_id !< The field ID to be added to field_ids
+  integer :: i !< For looping
+  do i = 1, size(obj%field_ids)
+    if (.not.obj%field_registered(i)) then
+      obj%field_ids(i) = new_field_id
+      obj%field_registered(i) = .true.
+      return
+    endif
+  enddo
```

Comment:
I think it is more efficient to add `number_of_registered_variables` to the file obj (or whatever you want to call it...)

```F90
if (obj%number_of_registered_variables + 1 .gt. size(obj%field_ids)) error out

obj%number_of_registered_variables = obj%number_of_registered_variables + 1
obj%field_ids (obj%number_of_registered_variables) = new_field_id
obj%field_registered(obj%number_of_registered_variables) = .true.
```

Reply:
`fms_diag_object` is managing the number of registered variables.  Is this a value for the number of registered variables just for a specific file?

Reply:
yes the number of fields that are registered for that file